compiler/src/vm/heap/mod.rs

```diff
@@ -0,0 +1,235 @@
+mod object;
+mod pointer;
+
+pub use self::{
+    object::{Builtin, Closure, Data, Int, Object, Struct, Symbol, Text},
+    pointer::Pointer,
+};
+use crate::builtin_functions::BuiltinFunction;
+use itertools::Itertools;
+use num_bigint::BigInt;
+use std::{cmp::Ordering, collections::HashMap};
+
+#[derive(Clone)]
+pub struct Heap {
+    objects: HashMap<Pointer, Object>,
+    next_address: Pointer,
+}
+
+impl std::fmt::Debug for Heap {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut objects = self.objects.clone().into_iter().collect_vec();
+        objects.sort_by_key(|(address, _)| address.raw());
```

Comment:
Do we really have to clone the whole heap or could we, e.g., collect and sort the keys and then access values in the `for` loop below?